common/wireaddr.c

```diff
@@ -461,7 +461,7 @@ bool parse_wireaddr_internal(const char *arg, struct wireaddr_internal *addr,
 		return false;
 
 	/* We can't do DNS, so keep unresolved. */
-	if (!wireaddr_from_unresolved(addr, ip, splitport)) {
+	if (!ip || !wireaddr_from_unresolved(addr, ip, splitport)) {
```

Comment:
Good catch!  This is a bug,  but this is the wrong fix AFAICT.

We should unconditionally call `separate_address_and_port` (and fail if that returns false, meaning the address is completely malformed).  Putting it under the wildcard_ok condition is a bug.